bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubWebSocket.java

```diff
@@ -154,85 +153,100 @@ private String jsonReplaceQuotes(String json) {
-            Response responseDto = gson.fromJson(responseOuter, Response.class);
-            if (responseDto == null) {
-                throw new JsonSyntaxException("Response DTO is invalid");
-            }
-            if (!NeoHubBindingConstants.HM_SET_COMMAND_RESPONSE.equals(responseDto.message_type)) {
-                throw new JsonSyntaxException("DTO 'message_type' field is invalid");
+            // don't throw an exception if already closing
+            if (closing) {
+                return "{}";
             }
-            String responseJson = responseDto.response;
-            if (responseJson == null) {
-                throw new JsonSyntaxException("DTO 'response' field is null");
+
+            // if an IOException was caught above, re-throw it again
+            caughtException = websocketException != null ? websocketException : caughtException;
+            if (caughtException != null) {
+                throw caughtException;
             }
-            responseJson = jsonUnEscape(responseJson).strip();
-            if (!JsonParser.parseString(responseJson).isJsonObject()) {
-                throw new JsonSyntaxException("DTO 'response' field is not a JSON object");
+
+            try {
+                Response responseDto = gson.fromJson(responseOuter, Response.class);
+                if (responseDto == null) {
+                    throw new JsonSyntaxException("Response DTO is invalid");
+                }
+                if (!NeoHubBindingConstants.HM_SET_COMMAND_RESPONSE.equals(responseDto.message_type)) {
+                    throw new JsonSyntaxException("DTO 'message_type' field is invalid");
+                }
+                String responseJson = responseDto.response;
+                if (responseJson == null) {
+                    throw new JsonSyntaxException("DTO 'response' field is null");
+                }
+                responseJson = jsonUnEscape(responseJson).strip();
+                if (!JsonParser.parseString(responseJson).isJsonObject()) {
+                    throw new JsonSyntaxException("DTO 'response' field is not a JSON object");
+                }
+                return responseJson;
+            } catch (JsonSyntaxException e) {
+                logger.debug("hub '{}' {}; response:{}", hubId, e.getMessage(), responseOuter);
+                throw new NeoHubException("Invalid response");
             }
-            return responseJson;
-        } catch (JsonSyntaxException e) {
-            logger.debug("hub '{}' {}; response:{}", hubId, e.getMessage(), responseOuter);
-            throw new NeoHubException("Invalid response");
+        } catch (InterruptedException e) {
+            throw new NeoHubException("Throttler was interrupted unexpectedly");
         }
     }
 
     @Override
     public void close() {
-        closeSession();
+        closing = true;
+        Session session = this.session;
+        if (session != null) {
+            session.close();
+        }
```

Comment:
No need to set this.session to null ?

Reply:
I believe that calling close() more than once on a session will not cause an error. However @lolodomo you are absolutely correct that it is better coding to null the session here. It will fix it.